src/python_pachyderm/mixin/pfs.py

```diff
@@ -707,3 +657,244 @@ def diff_file(self, new_commit, new_path, old_commit=None, old_path=None, shallo
+        converted to a header and applied to all file shards.
+        """
+        self._ops.append(AtomicPutFileobjOp(
+            commit, path, value,
+            delimiter=delimiter,
+            target_file_datums=target_file_datums,
+            target_file_bytes=target_file_bytes,
+            overwrite_index=pfs_proto.OverwriteIndex(index=overwrite_index) if overwrite_index is not None else None,
+            header_records=header_records,
+        ))
+
+    def put_file_from_bytes(self, commit, path, value, delimiter=None, target_file_datums=None,
+                            target_file_bytes=None, overwrite_index=None, header_records=None):
+        """
+        Uploads a PFS file from a bytestring.
+
+        Params:
+
+        * `commit`: A tuple, string, or `Commit` object representing the
+        commit.
+        * `path`: A string specifying the path in the repo the file(s) will be
+        written to.
+        * `value`: The file contents as a bytestring.
+        * `delimiter`: Optional. causes data to be broken up into separate
+        files with `path` as a prefix.
+        * `target_file_datums`: An optional int. Specifies the target number of
+        datums in each written file. It may be lower if data does not split
+        evenly, but will never be higher, unless the value is 0.
+        * `target_file_bytes`: An optional int. Specifies the target number of
+        bytes in each written file, files may have more or fewer bytes than
+        the target.
+        * `overwrite_index`: An optional int. This is the object index where
+        the write starts from.  All existing objects starting from the index
+        are deleted.
+        * `header_records: An optional int for splitting data when `delimiter`
+        is not `NONE` (or `SQL`). It specifies the number of records that are
+        converted to a header and applied to all file shards.
+        """
+        self.put_file_from_fileobj(
+            commit, path, io.BytesIO(value),
+            delimiter=delimiter,
+            target_file_datums=target_file_datums,
+            target_file_bytes=target_file_bytes,
+            overwrite_index=overwrite_index,
+            header_records=header_records,
+        )
+
+    def put_file_from_url(self, commit, path, url, delimiter=None, recursive=None, target_file_datums=None,
+                          target_file_bytes=None, overwrite_index=None, header_records=None):
+        """
+        Puts a file using the content found at a URL. The URL is sent to the
+        server which performs the request.
+
+        Params:
+
+        * `commit`: A tuple, string, or `Commit` object representing the
+        commit.
+        * `path`: A string specifying the path to the file.
+        * `url`: A string specifying the url of the file to put.
+        * `delimiter`: Optional. causes data to be broken up into separate
+        files with `path` as a prefix.
```

Comment:
An example would help clarify this for me (and in the other docstrings where `delimiter` is documented).